File: Fizzbuzz/Java/Fizzbuzz1/src/main/java/com/tonytroy/Fizzbuzz.java
package com.tonytroy;

import java.util.*;

public class Fizzbuzz {

    private static TreeMap<Integer, String> magicNumbers;

    public static String fizzbuzz(int size) {
        init();

        StringBuilder fizzbuzzNumbers = new StringBuilder();
        for(int i = 1; i <= size; i++){
            boolean isMagicNumber = false;
            for (int magicNumber : magicNumbers.descendingKeySet()) {
                if (i % magicNumber == 0) {
                    fizzbuzzNumbers.append(magicNumbers.get(magicNumber));
                    isMagicNumber = true;
                    break;
                }
            }

            if (!isMagicNumber)
                fizzbuzzNumbers.append(i);

            if (i < size) fizzbuzzNumbers.append(",");
        }

        return fizzbuzzNumbers.toString();
    }

    private static void init() {
        magicNumbers = new TreeMap<>();
        magicNumbers.put(15, "FizzBuzz");
        magicNumbers.put(5, "Buzz");
        magicNumbers.put(3, "Fizz");
    }
}

File: Fizzbuzz/python/try.03/fizzbuzz.py
def fizzbuzz(num):
    return [fizzbuzz_gen(x) for x in range(1, num + 1)]

def fizzbuzz_gen(num):
    ret_val = None

    if num % 3 == 0: ret_val = 'fizz'
    if num % 5 == 0: ret_val = 'buzz'
    if num % 15 == 0: ret_val = 'fizzbuzz'
    if ret_val is None: ret_val = num

    return ret_val
